### PR TITLE
filter the request if the request.body is invalid

### LINKED ISSUE #7: Giving an error with empty body
I'm using `koa-range` with `koa-better-static`, and the user is making a request with

```
if-modified-since:Fri, 01 Apr 2016 18:21:16 GMT
range:bytes=15110-6381395
```

`koa-better-static` returns `304 NOT MODIFIED` with an empty body because the file has not been modified since that date, but now `koa-range` tries to access `this.body.length` which throws an error as its dereferencing undefined

### PR DESCRIPTION
**Pull request by yorkie**
This fixes #7

### CODE REVIEW

**Comment by yorkie** [comment]
@koajs/body-parsing @coderhaoxin may I ask someone help me this review, thanks :-)

File: index.js [hunk 1]
```diff
@@ -25,7 +25,8 @@ module.exports = function * (next) {
 
   yield * next;
 
-  if (this.method != 'GET') {
+  if (this.method != 'GET' ||
+     this.body == undefined) {
```
Comment (coderhaoxin) [inline index.js]: perfer `this.body == null`, but NVM :)

**Comment by coderhaoxin** [comment]
LGTM

File: index.js (hunk 1 above)
Comment (yorkie) [inline index.js]: Good catch, thank you

**Pull request merged**